cacheops/query.py

```diff
@@ -38,14 +39,16 @@ def cache_thing(cache_key, data, cond_dnfs, timeout):
     """
     Writes data to cache and creates appropriate invalidators.
     """
-    load_script('cache_thing', CACHEOPS_LRU)(
-        keys=[cache_key],
-        args=[
-            pickle.dumps(data, -1),
-            json.dumps(cond_dnfs, default=str),
-            timeout
-        ]
-    )
+    # ignore writes while in transaction. no point in writing data that might be rolled back.
+    if not in_transaction():
+        load_script('cache_thing', CACHEOPS_LRU)(
+            keys=[cache_key],
+            args=[
+                pickle.dumps(data, -1),
+                json.dumps(cond_dnfs, default=str),
+                timeout
+            ]
+        )
```

Comment:
Should be used higher level like in `.iterator()`, `@cached_as()` and cache_on_save code. Just assert here.